keystore/kms/aws/keystore.go

```diff
@@ -0,0 +1,96 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	baseKMS "github.com/cossacklabs/acra/keystore/kms"
+)
+
+// Keystore implementation of AWS Keystore
+type Keystore struct {
+	cfg    *Configuration
+	client *KMSClient
+}
+
+// NewKeystore create new AWS KMS encryptor which implement Keystore interface
+func NewKeystore(credentialPath string) (baseKMS.Keystore, error) {
+	cfg, err := readConfigByPath(credentialPath)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := NewKmsClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &Keystore{cfg, client}, nil
+}
+
+// ID return source of
+func (k *Keystore) ID() string {
+	return "KMS AWS"
+}
+
+// CreateKey create key on KMS according to specification
+func (k *Keystore) CreateKey(ctx context.Context, metaData baseKMS.CreateKeyMetadata) (*baseKMS.KeyMetadata, error) {
+	keyMetadata, err := k.client.CreateKey(ctx, metaData)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := k.client.CreateAlias(ctx, *keyMetadata.Arn, metaData.KeyName); err != nil {
+		return nil, err
+	}
+
+	return &baseKMS.KeyMetadata{
+		KeyID: *keyMetadata.Arn,
+	}, nil
+}
+
+// IsKeyExist check if key is present on KMS
+func (k *Keystore) IsKeyExist(ctx context.Context, keyID string) (bool, error) {
+	aliases, err := k.client.ListAliases(ctx, nil)
+	if err != nil {
+		return false, err
+	}
+
+	for _, alias := range aliases {
+		// format of ARN is arn:aws:kms:eu-west-1:account-id:alias/acra-master-key-test
+		arnParts := strings.Split(*alias.AliasArn, ":")
+		if len(arnParts) < 6 {
+			return false, errors.New("invalid arn format found for keyId")
+		}
+		region := arnParts[3]
+
+		if *alias.AliasName == fmt.Sprintf("alias/%s", keyID) && region == k.cfg.Region {
```

Comment:
much better to create new function like `getKeyName(keyID string) string {return "alias/"+keyID}`. Because it will be much faster than `fmt.Sprintf` which need to parse string at first and after that place value for placeholder.